Settings: add new card for managing plugins.

jeherve opened the pull request:
Fixes #10694 

#### Changes proposed in this Pull Request:

This PR adds a new card for managing plugins.

**When connected**
<img width="904" alt="screenshot 2018-12-10 at 22 36 23" src="https://user-images.githubusercontent.com/426388/49764315-605df400-fccf-11e8-85f5-dafbc0012fd9.png">

**When in dev mode**

<img width="845" alt="screenshot 2018-12-10 at 22 49 55" src="https://user-images.githubusercontent.com/426388/49764326-65bb3e80-fccf-11e8-91de-fdcaa3715419.png">

#### Testing instructions:

* Go to Jetpack > Settings while your site is in dev mode, with Manage active, or with Manage off.
* Make sure the new card in the Security tab looks good.

#### Proposed changelog entry for your changes:
* Settings: add new card for plugins management.

jeherve committed:
Settings: add new card for managing plugins.

Fixes #10694

jeherve committed:
Admin page: update new component to use new naming convention.

@see #10932

oskosk commented:
This one also needs a rebase to help 0c1819c9965b4de8699fe7c1842470f048606ef7

oskosk reviewed (dismissed):
Good to merge after rebasing!

Note that manage being on or off (mentioned in the testing instructions) does not make a difference in the aspect of this new card.

jeherve commented:
I rebased. Ready for a design review before merge, maybe?

keoshi commented:
@jeherve From now on, it would be awesome if you could add the entire @Automattic/jetpack-design group on the PR review. I've asked Derek to add this yesterday, so everyone can consistently ping us and get a faster review hopefully. Will review right away — thanks!

oskosk commented on _inc/client/security/manage-plugins.jsx:
```diff
@@ -0,0 +1,75 @@
+		}
+
+		render() {
+			const configLink = () => {
```
Can we make `configLink` be a class method instead ?

jeherve replied:
Done in e2bd16022cabef9f9f0c111c9c62093d4e3b44bc

oskosk reviewed (dismissed):
Sorry, I want to propose something

jeherve committed:
Move config link to a class method

@see https://github.com/Automattic/jetpack/pull/10924#discussion_r240695753

keoshi reviewed (commented):
Works great in my testing! One thing I'd like to address before we merge if possible — the corresponding card on the dashboard looks different:

![image](https://user-images.githubusercontent.com/390760/49816964-f55ffc00-fd66-11e8-9d81-c7c068860da8.png)

- Dashboard card: replace chevron/arrow with external icon.
- Dashboard card: open link in new tab/window.
- Settings card: change label to “Manage your plugins”.

jeherve committed:
Settings card: change label to "Manage your plugins".

jeherve committed:
Dashboard card: update to make it look more similar to settings

- replace chevron/arrow with external icon.
- open link in new tab/window.
- run prettier

oskosk reviewed (approved):
LGTM!
